src/Ad.ts

```diff
@@ -215,10 +205,21 @@ class Ad implements IAd {
     // Instantiate all class based plugins and reference them
     this.attachPlugins(plugins);
 
-    this.callPlugins('onCreate');
+    const executionOfPlugins = this.callPlugins('beforeCreate');
 
     this.onReady(
-      () => this.callPlugins('afterCreate'),
+      async () => {
+        await executionOfPlugins;
+
+        await Promise.all(
```

Comment:
This should not be a promise.all. It will be far too brittle for the use case.
I will create a work item to address it in a following pr

Reply:
👍 Good catch